lib/data/odata-filter.js

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/opendatakit/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const odataParser = require('odata-v4-parser');
+const Problem = require('../util/problem');
+
+////////////////////////////////////////
+// AST NODE TRANSFORMATION
+
+const extractFunctions = [ 'date', 'time', 'year', 'month', 'day', 'hour', 'minute', 'second' ];
```

Comment:
Ah I didn't think about this during review, but I'm wondering now whether Postgres `extract()` supports `date` or `time`. (I don't see them in the list here: https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT.)

Reply:
okay it looks weird and complicated i'll just take em out.
]

Reply:
okay it isn't that complicated but also actually i dunno what good those funcs are rn.